src/sentry/options/defaults.py

```diff
@@ -1305,24 +1305,29 @@
 register("span_descs.bump-lifetime-sample-rate", default=0.25, flags=FLAG_AUTOMATOR_MODIFIABLE)
 # Decides whether artifact bundles asynchronous renewal is enabled.
 register("sourcemaps.artifact-bundles.enable-renewal", default=0.0, flags=FLAG_AUTOMATOR_MODIFIABLE)
-# Enables reporting status of queues for backpressure management.
-register(
-    "backpressure.monitor_queues.enable_status", default=False, flags=FLAG_AUTOMATOR_MODIFIABLE
-)
-# Enables checking queue health in consumers for backpressure management.
-register("backpressure.monitor_queues.enable_check", default=False, flags=FLAG_AUTOMATOR_MODIFIABLE)
+
+# === Backpressure related runtime options ===
+
+# Enables monitoring of services for backpressure management.
+register("backpressure.monitoring_enabled", default=False, flags=FLAG_AUTOMATOR_MODIFIABLE)
+# How often the monitor will check service health.
+register("backpressure.monitoring_interval", default=5, flags=FLAG_AUTOMATOR_MODIFIABLE)
+
+# Enables checking consumer health for backpressure management.
+register("backpressure.checking_enabled", default=False, flags=FLAG_AUTOMATOR_MODIFIABLE)
+# How often a consumer will check for its health in a debounced fassion.
+register("backpressure.checking_interval", default=5, flags=FLAG_AUTOMATOR_MODIFIABLE)
+
+# The high-watermark levels per-service which will mark a service as unhealthy.
+# This should mirror the `SENTRY_PROCESSING_SERVICES` setting.
 register(
-    "backpressure.monitor_queues.check_interval_in_seconds",
-    default=5,
+    "backpressure.high_watermarks",
+    default={
+        "celery": 0.5,
+        "attachments-store": 0.5,
+        "processing-store": 0.5,
+        "processing-locks": 0.5,
+        "post-process-locks": 0.5,
+    },
```

Comment:
@beezz could you please check if a structured option like this one is convenient for you? Otherwise, we might need to split this into smaller options.

Reply:
The issue with such an option is that if you add a new key to the default value, you must update each environment where overrides are in place.

Reply:
@Swatinem The above is a good point for why we should still split up this option.